src/main/java/de/yuna/berlin/nativeapp/services/http/model/HttpRequest.java

```diff
@@ -0,0 +1,250 @@
+            return null;
+        }
+    }
+
+    public TypeContainer bodyAsJson() {
+        return JsonDecoder.jsonTypeOf(bodyAsString());
+    }
+
+    public TypeMap queryParameters() {
+        if (queryParams == null) {
+            try {
+                String query = exchange.getRequestURI().getQuery();
+                queryParams = new TypeMap();
+                if (query != null) {
+                    String[] queryParamsArray = query.split("&");
+                    for (String param : queryParamsArray) {
+                        String[] keyValue = param.split("=");
+
+                        if (keyValue.length == 2) {
+                            String key = java.net.URLDecoder.decode(keyValue[0], StandardCharsets.UTF_8);
+                            String value = java.net.URLDecoder.decode(keyValue[1], StandardCharsets.UTF_8);
+                            queryParams.put(key, value);
+                        } else if (keyValue.length == 1) {
+                            String key = java.net.URLDecoder.decode(keyValue[0], StandardCharsets.UTF_8);
+                            queryParams.put(key, "");
+                        }
+                    }
+                }
+            }catch (Exception ignored) {
+            }
+        }
+        return queryParams;
+    }
+
+    public boolean containsQueryParam(String key) {
+        return queryParams.containsKey(key);
+    }
+
+    public String getQueryParam(String key) {
+        return queryParams.get(key).toString();
+    }
+
+
+    public boolean exactMatch(String path) {
+        return this.path.startsWith(path);
+    }
+
+    public boolean match(String pathToMatch) {
+
+        String[] partsToMatch = pathToMatch.split("/");
+        String[] parts = path.split("/");
+
+        if (exactMatch(pathToMatch) && pathToMatch.contains("{"))
+            return true;
+
+        if (partsToMatch.length != parts.length)
+            return false;
+
+        if (pathParams == null) {
+            pathParams = new TypeMap();
+            for (int i = 0; i < partsToMatch.length; i++) {
+                if (!partsToMatch[i].equals(parts[i])) {
+                    if (partsToMatch[i].startsWith("{")) {
+                        pathParams.put(partsToMatch[i], parts[i]);
+                    }else {
+                        return false;
+                    }
+                }
+            }
+        }
+
+        return true;
+    }
+
+
+    public TypeMap getPathParams() {
+        return pathParams;
+    }
+
+    public String getHeader(String name) {
+        return headers.get(name).toString();
+    }
+
+    public TypeMap getHeaders() {
+        return headers;
+    }
+
+    public void addHeader(String name, String value) {
+        headers.put(name, value);
+    }
+
+    public void removeHeader(String name) {
+        headers.remove(name);
+    }
+
+    public boolean containsHeader(String name) {
+        return headers.containsKey(name.toLowerCase());
+    }
+
```

Comment:
add more methods:

* `scheme()` or `protocol` which returns "http", "https"....
* `fragment()` which returns the fragment after `#`
* `host()` which returns "example.com" or the ip address
* `port()`